File: src/Services/Employee/TalentMesh.EmployeeService/Models/Employee.cs
namespace TalentMesh.EmployeeService.Models;

public class Employee
{
    public Guid Id { get; set; }

    public string Name { get; set; } = string.Empty;

    public string Email { get; set; } = string.Empty;

    public string Department { get; set; } = string.Empty;

    public string Designation { get; set; } = string.Empty;

    public decimal ExperienceYears { get; set; }

    public string Location { get; set; } = string.Empty;

    public int AllocationPercentage { get; set; }

    public ICollection<EmployeeSkill> Skills { get; set; }
        = new List<EmployeeSkill>();
}

File: src/Frontend/TalentMesh-UI/src/app/models/employee.ts

export interface Skill {
  id: string;
  name: string;
  category: string;
}

export interface EmployeeSkill {
  id: string;
  level: string;
  score: number;
  yearsOfExperience: number;
  skill: Skill;
}

export interface Employee{
    id : string;
    name : string;
    designation : string;
    experienceYears : string;
    location : string;
    allocationPercentage:  number;
    skills : EmployeeSkill[]
}